spring-beans/src/main/java/org/springframework/beans/factory/support/ConstructorResolver.java

```diff
@@ -771,7 +771,7 @@ private ArgumentsHolder createArgumentArray(
 							methodParam, beanName, autowiredBeanNames, converter, fallback);
 					args.rawArguments[paramIndex] = autowiredArgument;
 					args.arguments[paramIndex] = autowiredArgument;
-					args.preparedArguments[paramIndex] = new AutowiredArgumentMarker();
+					args.preparedArguments[paramIndex] = AutowiredArgumentMarker.INST;
```

Comment:
If you use the singleton here, that would be OK, considering my other comments below.